src/core/mavsdk_impl.cpp

```diff
@@ -594,8 +594,8 @@ void MavsdkImpl::send_heartbeat()
         get_mav_type(),
         get_own_component_id() == MAV_COMP_ID_AUTOPILOT1 ? MAV_AUTOPILOT_GENERIC :
                                                            MAV_AUTOPILOT_INVALID,
-        0,
-        0,
+        _base_mode,
+        _custom_mode,
```

Comment:
Oh, this is for when Mavsdk is an autopilot :thinking:.

What about only setting this if `get_own_component_id() == MAV_COMP_ID_AUTOPILOT1`?